SYCL/ESIMD/private_memory_access.cpp

```diff
@@ -0,0 +1,179 @@
+//==--------------- private_memory_access.cpp - DPC++ ESIMD on-device test -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out 1
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out 2
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out 3
+
+// Since in ESIMD a single WI occupies entire underlying H/W thread, SYCL
+// private memory maps to what's known as 'thread private memory' in CM.
+// This test is intended to check TPM support implementation in ESIMD
+// backend. In order to force using of TPM need to allocate 96x32 bytes or more.
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/esimd.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+
+constexpr unsigned VL = 8;
+constexpr unsigned SZ = 800; // big enough to use TPM
+
+ESIMD_INLINE void work(int *o, int case_num, int offx1, int offx2, int offy1,
```

Comment:
IMO, it is better not to pollute test cases with unrelated code. DCE on `if`s with compile-time evaluated conditions should help. Suggestions below implement that.

```suggestion
ESIMD_INLINE template<int CASE_NUM> void work(int *o, int offx1, int offx2, int offy1,
```